### ISSUE TITLE
maybe need to flush? when cli pygmentize without -s and with colorma

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#860]
(Original issue 1153 created by upgrade_ayp on 2015-10-01T18:13:09.717166+00:00)

when I tried this code on iPython Notebook,
```
#!python
!git show master:some_file.js | pygmentize -l js -f terminal
```
output code was half finished.
My env is Win8.1 and Python 3.4.3

I added some codes to cmdline.py as below
```
#!python

    # ... and do it!
    if '-s' not in opts:
        # process whole input as per normal...
        highlight(code, lexer, fmter, outfile)
        if hasattr(outfile, 'flush'):
            outfile.flush()
        return 0
```
this code fixed my problem.
I'm a newbie in python, so i don't know the details..
please look it up. thanx

**Comment by Anteru** [comment]
(Original issue was assigned to tshatch)

(Original issue 1153 last updated on 2015-10-27T15:08:23.900248+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by tshatch on 2015-10-17T01:16:39.268648+00:00)

I have a suspicion this is because of some hacks in pygments/util.py:

```
#!python

    class UnclosingTextIOWrapper(TextIOWrapper):
        # Don't close underlying buffer on destruction.
        def close(self):
            pass

```

(if you change the 'pass' to 'self.flush()' does it work?)

**Comment by Anteru** [comment]
(Original comment by upgrade_ayp on 2015-10-24T22:24:27.004270+00:00)

sorry for my late reply.

> (if you change the 'pass' to 'self.flush()' does it work?)

I reverted my hack, and tried `self.flush()` that you taught me.
your solution works fine without fail.

**Comment by Anteru** [comment]
(Original comment by tshatch on 2015-10-27T15:08:23.870945+00:00)

Fixes #1153: Change Windows+Py3 workaround code to flush when finished.

This avoids a case where the last chunk might not be output in iPython Notebook on Windows.

→ <<cset 09e64436b4c1>>
